### ISSUE TITLE
Cannot follow an account that has ever been private or previously followed

### ISSUE THREAD

**Issue by dnno** [pixelfed/pixelfed#866]
I have an account at pixelfed.social and I'd like to follow it from my mastodon account, but it doesn't work.  It says something like "sent following request".

My pixelfed account was private previously, but is public now. I don't get an approval request on pixelfed either.

I was.able to follow my account in the past, but now I can't.

I understand that federation is not necessarily completely done and without errors, yet. Is this a known problem? 

Can I provide additional information?

**Comment by dansup** [comment]
I can't reproduce this issue, I successfully followed you from mastodon.social.

**Comment by asonix** [comment]
I also have this issue with my instance, pooltoy.club. When following from Mastodon, the initial response will make it look as if you are following the pixelfed account, but if you check back later, it will be the follow requested icon. My pixelfed account recognizes that it has been followed from Mastodon, though

**Comment by dnno** [comment]
Same here, my mastodon account indeed is showing up in the followers of my pixelfed account. But on the mastodon instance it's not recognized...

**Comment by trwnh** [comment]
It would help to know the exact accounts/domains being used. I personally have no issue following my pixelfed.social test account from mastodon.social (and seeing my pixelfed posts in mastodon, as well).

**Comment by asonix** [comment]
following pooltoy.club/asonix (pixelfed) from asonix.dog/@asonix (mastodon) has never properly worked for me. There don't seem to be any errors in any logs, so I'm not sure what's going on there

**Comment by trwnh** [comment]
This is what I'm seeing:
1) From trwnh@mastodon.social, attempt following asonix@pooltoy.club
2) Follow button changes to Unfollow
3) Followers count goes from 7 to 8
4) Some seconds (15-30) later: Follower count goes from 8 to 7, button still in Unfollow state
5) Refresh page
6) Unfollow button has changed into "Awaiting approval"

So my guess would be something to do with the way `Accept {Follow}` is sent out. More specifically, there seems to be a bug in pooltoy.club's deployment or in the asonix account, because following a pixelfed.social account has previously worked just fine (and I am still following my trwnh@pixelfed.social), and moreover it continues to work (as I have successfully followed dansup@pixelfed.social just now).

EDIT: I have just created trwnh@pooltoy.club for testing your instance, and was able to successfully follow that account as well. So the issue is definitely tied to specific accounts and not the entire instance. (Side note: I was able to follow the account despite not even receiving the verification email.)

**Comment by trwnh** [comment]
I think I've isolated the issue to being due to incorrect unprivating of an account, and lack of state sync between two servers. I am unable to verify this 100% because I don't know how to block accounts that are following a pixelfed account, so I am unable to softblock in order to force an unfollow. Regardless, here are my steps:

1) Sign up as trwnh@pixfed.com
2) Set account to private
3) Send follow request from trwnh@mastodon.social successfully
4) Accept follow request in pixfed.com notifications
5) Observe that both mastodon.social and pixfed.com are successfully aware of the follow relation
6) Make a test post in trwnh@pixfed.com
7) Test post not visible from mastodon.social
8) Unfollow trwnh@pixfed.com
9) Set account to public
10) Unable to refollow trwnh@pixfed.com; pixfed.com shows that trwnh@mastodon.social is still following (despite having unfollowed)

In summary, I am only able to follow public accounts once. After that, behavior becomes unpredictable.

EDIT: Re-tested my hypothesis. I am now unable to follow dansup@pixelfed.social despite having successfully followed and unfollowed that account just 10 minutes ago. So what needs to be done:

- [ ] Process incoming Undo Follow / Reject Follow to remove an account from your followers
- [ ] Start again sending automatic outgoing Accept Follow for private accounts that have been changed to public, maybe?

**Comment by dansup** [comment]
@trwnh I am investigating. I recently added [Undo Follow](https://github.com/pixelfed/pixelfed/blob/dev/app/Util/ActivityPub/Inbox.php#L342-L345) and it already sends automatic [Accept Follow](https://github.com/pixelfed/pixelfed/blob/dev/app/Util/ActivityPub/Inbox.php#L174-L215) requests.
